webviz_subsurface/containers/_summary_stats.py

```diff
@@ -7,41 +7,69 @@
-statistical plots (min, max, mean, p10, p90).
+    """Summary statistics
 
-* `ensemble`: Which ensemble in `container_settings` to visualize.
-* `sampling`: Optional. Either `monthly` or `yearly`. Default is `monthly`.
-* `title`: Optional title for the container.
-'''
+    Args:
+        ensembles: list of ensemble paths. Length >= 1
+        column_keys: list of pre defined vectors to visualize. Default is none
+        sampling: Optional. Either 'monthly' or 'yearly'. Default is 'monthly'.
+        title: Optional title for the container.
+        history_uncertainty: boolean value if history vector is subjected to
+            uncertainty.
+    Return:
+        dcc.Graph() of either statistics plot or realization plot.
+    Annotations:
+        Function argument-litst are passed as tuple as they are hashed in
+        webviz-store --portable option (list-type is not hashable).
+        Pandas within fmu.ensemlbe expects list. Therefore argument-tuples are
+        converted back into lists within get_smry_data and get_smry_stats
+        functions.
+    """
 
     def __init__(
             self,
             app,
             container_settings,
-            ensemble,
+            ensembles,
+            column_keys=None,
             sampling: str = 'monthly',
-            title: str = 'Simulation time series'):
+            title: str = 'Simulation time series',
+            history_uncertainty: bool = False):
 
         self.title = title
         self.dropwdown_vector_id = 'dropdown-vector-{}'.format(uuid4())
+        self.column_keys = tuple(column_keys) if isinstance(
+            column_keys, (list, tuple)) else None
         self.sampling = sampling
         self.radio_plot_type_id = 'radio-plot-type-{}'.format(uuid4())
+        self.show_history_uncertainty_id = 'show-history-uncertainty-{}'.format(
+            uuid4())
         self.chart_id = 'chart-id-{}'.format(uuid4())
-
-        # Finding all summary vectors:
-        self.ensemble_path = container_settings['scratch_ensembles'][ensemble]
-
-        self.smry_columns = sorted(list(get_summary_data(self.ensemble_path,
-                                                         self.sampling)
-                                        .drop(columns=['DATE', 'REAL'])
-                                        .columns))
-
+        self.ensemble_paths = tuple(
+            (ensemble,
+             container_settings['scratch_ensembles'][ensemble])
+            for ensemble in ensembles)
+        self.smry_columns = sorted(
+            list(
+                get_summary_data(
+                    ensemble_paths=self.ensemble_paths,
+                    sampling=self.sampling,
+                    column_keys=self.column_keys) .drop(
+                    columns=[
+                        'DATE',
+                        'REAL',
+                        'ENSEMBLE']) .columns))
+        self.smry_vector_columns = tuple([col for col in self.smry_columns
```

Comment:
`col_endswith("H"'`) is not correct as this doesn't differentiate vectors e.g. per well.
E.g. the vector `GGOR:WELLH` should not be filtered.
=> One solution could be to split on `:` and filter on the first element only

Reply:
Pseudocode now:
    lstH = [i + 'H' for i in vctrs if i + 'H' in vctrs]
    lst = [i for i in vctrs if i not in lstH]